### PR TITLE
Fully integrate alias domains

### PR DESCRIPTION
**Pull request by mundschenk-at**
* Support log parsing and displaying graphs for alias domains.
* Undo SRS rewriting for better domain detection.

The PR also adds code to explicitly ignore unknown RRD data and adds tags for all debug messages. Both of these commits are probably not strictly necessary, so I'm open to dropping them.

### CODE REVIEW

**Comment by mundschenk-at** [comment]
@tonioo I'll continue working on the other issues/PRs once this one has been successfully merged.

File: modoboa_stats/management/commands/logparser.py [hunk 1]
```diff
@@ -277,73 +283,164 @@ def year(self, month):
             return self.__year - 1
         return self.__year
 
-    def _parse_line(self, line):
-        """Parse a single log line.
+    def domain(self, mail_address):
```
Comment (tonioo) [inline modoboa_stats/management/commands/logparser.py]: There is already something available in modoboa to split an email address: https://github.com/modoboa/modoboa/blob/master/modoboa/lib/email_utils.py#L221.

File: modoboa_stats/management/commands/logparser.py [hunk 2]
```diff
@@ -277,73 +283,164 @@ def year(self, month):
...
+    def is_srs_forward(self, mail_address):
+        """ Check if mail address has been mangled by SRS
+
+        Sender Rewriting Scheme (SRS) modfies mail adresses so that they
+        always end in a local domain.
+
+        :param str mail_address
+        :return a boolean
+        """
+        return re.match("^(?:srs|SRS)[01]", mail_address) != None
```
Comment (tonioo) [inline modoboa_stats/management/commands/logparser.py]: `is not None`

File: modoboa_stats/management/commands/logparser.py [hunk 3]
```diff
@@ -277,73 +283,164 @@ def year(self, month):
...
+
+    def reverse_srs(self, mail_address):
+        """ Try to unwind a mail address rewritten by SRS
+
+        Sender Rewriting Scheme (SRS) modfies mail adresses so that they
```
Comment (tonioo) [inline modoboa_stats/management/commands/logparser.py]: modifies

File: modoboa_stats/views.py [hunk 4]
```diff
@@ -66,12 +68,12 @@ def check_domain_access(user, pattern):
             return domains.first().name
         return "global"
 
-    results = Domain.objects.filter(name__startswith=pattern)
-    if results.count() != 1:
+    results = list(chain(Domain.objects.filter(name__startswith=pattern), DomainAlias.objects.filter(name__startswith=pattern)))
```
Comment (tonioo) [inline modoboa_stats/views.py]: This is not PEP8 compliant: > 80 cols ;)

File: modoboa_stats/management/commands/logparser.py [hunk 5]
```diff
@@ -277,73 +283,164 @@ def year(self, month):
...
+        host, prog, pid, log = m.groups()
+
+        if self._parse_amavis(prog, log, host, pid):
             return
-        self._dprint("Unknown line format: %s" % line_log)
+
+        if self._parse_postfix(prog, log, host, pid):
+            return
+
+        self._dprint("[parser] unknown line format: %s" % log)
```
Comment (tonioo) [inline modoboa_stats/management/commands/logparser.py]: You could rewrite this part like this:
```python
try:
    getattr(self, "_parse_{}".format(prog))(log, host, pid)
except AttributeError:
    self._dprint("[parser] unknown line format: {}".format(log))
```

**Comment by mundschenk-at** [comment]
@tonioo Regarding single quotes: I'll replace them, I just didn't want to touch existing lines only for that reason.

**Comment by tonioo** [comment]
@mundschenk-at ok thanks. Have you seen the other comments I made?

**Comment by mundschenk-at** [comment]
@tonioo Yes. I'll replace the `domain` method with `split_mailbox` tonight. I didn't get around to it yesterday because the return signature necessitates a few more changes. All others are in, except for the `getattr` call. While very elegant, I'm not totaly convinced. There might be routines that should not depend the `prog` (i.e. "try with this method, if it fails, try another one")? Or maybe that's too complicated thinking.

**Comment by tonioo** [comment]
According to your current code, my proposal is fine. If you want a "try and guess" mechanism, then it won't work.

**Comment by mundschenk-at** [comment]
Yeah, I was wondering if we might need such a mechanism. But then again, the parser is looking at individual lines, so having a "try and guess" mechanism is of doubtful usability. Also, a "guessing" parser function could just call another method if necessary.

**Comment by tonioo** [comment]
@mundschenk-at great! Thank you for your work.

**Pull request merged**